### DISCUSSION TITLE
Adding watchers to an issue

### DISCUSSION THREAD

**Discussion by MattPD** [ankitpokhrel/jira-cli#452] [Q&A]
Is it possible to add watchers to an issue?

Following https://github.com/ankitpokhrel/jira-cli/discussions/346 I've tried these:
```
jira issue edit ISSUE-123 --custom="Watchers=user1@hostname1,user2@hostname2" --no-input
jira issue edit ISSUE-123 --custom Watchers="user1@hostname1,user2@hostname2" --no-input
```
Although `✓ Issue updated` is displayed it appears to have no effect (no watchers are added to the issue).

**Comment by ankitpokhrel** [comment]
Is `Watchers` a custom field in your installation? 🤔 

Its not a custom field as far as I can tell so you won't be able to do the above. It says `✓ Issue updated` because the jira API will ignore that field when updating the issue. This feature needs to be separately implemented.

![Screen Shot 2022-09-11 at 9 23 15 AM](https://user-images.githubusercontent.com/2364546/189516766-e5e540f6-f7c2-499f-abe8-e16e260804fa.png)

**Comment by MattPD** [comment]
FWIW, I can see it in among the createmeta fields:
```
curl -k --request GET \
  --url "${JIRA_SERVER?}/rest/api/2/issue/createmeta?projectKeys=PE&expand=projects.issuetypes.fields" \
  --user "<user>:${JIRA_API_TOKEN?}"
```
output contains
```
"customfield_10305":{"required":false,"schema":{"type":"array","items":"user","custom":"com.burningcode.jira.issue.customfields.impl.jira-watcher-field:watcherfieldtype","customId":10305},"name":"Watchers","fieldId":"customfield_10305","autoCompleteUrl":"https://JIRA_SERVER/rest/api/1.0/users/picker?fieldName=customfield_10305&query=","hasDefaultValue":false,"operations":["add","set","remove"]}
```

**Comment by yyamaguchi93** [comment]
any updates to this issue? running into the same.
